File: exercism/go/darts/darts.go
package darts

func Score(x, y float64) int {
	rsq := x*x + y*y // avoid sqrt
	switch {
	case rsq <= 1.0*1.0:
		return 10
	case rsq <= 5.0*5.0:
		return 5
	case rsq <= 10.0*10.0:
		return 1
	}
	return 0
}

File: exercism/c/darts/darts.c
#include "darts.h"

uint8_t score(coordinate_t c) {
  float dist_sq = c.x * c.x + c.y * c.y;
  if (dist_sq <= 1)
    return 10;
  if (dist_sq <= 25)
    return 5;
  if (dist_sq <= 100)
    return 1;
  return 0;
}
